src/shogun/lib/DynamicObjectArray.h

```diff
@@ -35,153 +37,52 @@ class CDynamicObjectArray : public CSGObject
 		CDynamicObjectArray()
 		: CSGObject(), m_array()
 		{
-			dim1_size=1;
-			dim2_size=1;
-			dim3_size=1;
-
+			dim1_size = 1;
```

Comment:
I am slightly missing the point of having only 2 dimensions now. Could you explain?

Reply:
DynProg.cpp ... shitty stuff. :)
on the end of the day if we could use some we could drop this whole thing and have std::vector<Some<CSGObject>> and not have this whole DynamicObjectArray...

Reply:
i could keep the dim3 but never used and since on the end of the day we really would like to deprecate this object (see std::vector<Some<CSGobject>>)